packages/database/src/TupaiaDatabase.js

```diff
@@ -268,11 +268,15 @@ export class TupaiaDatabase {
   async createMany(recordType, records) {
     // generate ids for any records that don't have them
     const sanitizedRecords = records.map(r => (r.id ? r : { id: this.generateId(), ...r }));
-    await this.query({
-      recordType,
-      queryMethod: QUERY_METHODS.INSERT,
-      queryMethodParameter: sanitizedRecords,
-    });
+    const batchSize = this.maxBindingsPerQuery;
+    for (let i = 0; i < sanitizedRecords.length; i += batchSize) {
+      const batchOfRecords = sanitizedRecords.slice(i, i + batchSize);
```

Comment:
True! I think you're remembering `executeSqlInBatches`, which does that for raw sql. It would make sense to extract the common util. Cheers for the suggestion :+1:

Reply:
This made a few places much cleaner!